views/metrilo-import-view.php

```diff
@@ -1,116 +1,147 @@
+                        import_type(newChunkId, dataOptions, 'orders', null);
+                        break;
+                    default:
+                        return false;
                 }
+            });
+        }
 
-                function import_type(newChunkId, data, current, next) {
-                    if (data[`${current}Chunks`] > 0) {
-                        data.percentage = (100 / data[`${current}Chunks`]);
-                    }
-                    var hasMoreChunks = newChunkId < data[`${current}Chunks`];
+        function ajax_post_with_retry(url, data, callback) {
+            if (dataOptions.retryAttempts !== 0) {
+                jQuery.post(url, data, function (response) {
+                    callback();
+                }).fail(function () {
+                    dataOptions.retryAttempts -= 1;
+                    setTimeout(function () {
+                        ajax_post_with_retry(url, data, callback);
+                    }, 5000);
+                })
+            } else {
+                dataOptions.retryAttempts = 3;
+                sync_chunk(data.chunkId + 1, data.importType, true);
+            }
+        }
 
-                    if(hasMoreChunks) {
-                        this.sync_chunk(newChunkId, data.importType, false);
-                    } else {
-                        if(current == 'orders') {
-                            this.update_importing_message("<span style='color: green;'>" + 'Done! Please expect up to 30 minutes for your historical data to appear in Metrilo.' + "</span>", true);
-                            jQuery('#metrilo_import_step').hide();
-                        } else {
-                            jQuery('#metrilo_import_step').text((`Importing ${next}`));
-                            dataOptions.importType = next;
-                            this.sync_chunk(0, dataOptions.importType, false);
-                        }
-                    }
-                }
+        function import_type(newChunkId, data, current, next) {
+            if (data[`${current}Chunks`] > 0) {
+                data.percentage = (100 / data[`${current}Chunks`]);
+            }
+            var hasMoreChunks = newChunkId < data[`${current}Chunks`];
 
-                function update_importing_message(message, show_loader) {
-                    if (show_loader) {
-                        jQuery('#metrilo_import_status').html(message);
-                    }
+            if (hasMoreChunks) {
+                sync_chunk(newChunkId, data.importType, false);
+            } else {
+                if (current == 'orders') {
+                    update_importing_message("<span class='metrilo-color-green'>" + 'Done! Please expect up to 30 minutes for your historical data to appear in Metrilo.' + "</span>", true);
+                    jQuery('#metrilo_import_step').hide();
+                } else {
+                    jQuery('#metrilo_import_step').text((`Importing ${next}`));
```

Comment:
I think you can extract fetching the next import type to a function, something like:

```
fetch_next_import(current) {
 array = { 'customers' => 'categories', 'categories' => 'deletedProduct' ... };
 return array[current];
} 
```

Then the big `switch` statement above will turn to a single method invokation:
```
import_type(newChunkId, dateOptions, import_type)
```
since you will be fetching the next import inside the `import_type` function and you won't pass it as an argument. 

It's just a suggestion and we can discuss in person if it doesn't make sense as a PR comment.

Reply:
I could not implement it the way you suggest. The script gets buggy this way.

Reply:
you know best :+1: